File: AdminApi.Sdk/Endpoints/Identities/Types/IdentityOverview.cs
using Backbone.AdminApi.Sdk.Endpoints.Tiers.Types;

namespace Backbone.AdminApi.Sdk.Endpoints.Identities.Types;

public class IdentityOverview
{
    public required string Address { get; set; }
    public required DateTime CreatedAt { get; set; }
    public DateTime? LastLoginAt { get; set; }
    public string? CreatedWithClient { get; set; }
    public int? DatawalletVersion { get; set; }
    public required byte IdentityVersion { get; set; }
    public Tier? Tier { get; set; }
    public int? NumberOfDevices { get; set; }
}

File: Applications/AdminUi/packages/admin_api_types/lib/src/identities/identity_overview.dart
import 'package:json_annotation/json_annotation.dart';

import '../tiers/tier.dart';

part 'identity_overview.g.dart';

@JsonSerializable()
class IdentityOverview {
  final String address;
  final DateTime createdAt;
  final String createdWithClient;
  final int identityVersion;
  final int numberOfDevices;
  final Tier tier;
  final DateTime? lastLoginAt;
  final int? datawalletVersion;

  IdentityOverview({
    required this.address,
    required this.createdAt,
    required this.createdWithClient,
    required this.identityVersion,
    required this.numberOfDevices,
    required this.tier,
    this.lastLoginAt,
    this.datawalletVersion,
  });

  factory IdentityOverview.fromJson(dynamic json) => _$IdentityOverviewFromJson(json as Map<String, dynamic>);
  Map<String, dynamic> toJson() => _$IdentityOverviewToJson(this);
}
